akka-cluster-sharding/src/main/resources/reference.conf

```diff
@@ -116,6 +122,18 @@ akka.cluster.sharding {
     max-simultaneous-rebalance = 3
   }
 
+  # Setting for akka.cluster.sharding.ShardCoordinator.LeastShardAllocationStrategy2
+  least-shard-allocation-strategy2 {
+    # Maximum number of shards that will be rebalanced in one rebalance round.
+    # The lower of this and `relative-limit` will be used.
+    absolute-limit = 100
+
+    # Maximum number of shards that will be rebalanced in one rebalance round.
+    # Fraction of total number of (known) shards.
+    # The lower of this and `absolute-limit` will be used.
+    relative-limit = 0.1
+  }
```

Comment:
How do we plan to evolve this settings? Will we carry around `least-shard-allocation-strategy2` forever? 

What about...

```
least-shard-allocation-strategy {

  impl = "legacy" # 'legacy' or 'new-gen'

  # settings for legacy strategy
  rebalance-threshold = 1
  max-simultaneous-rebalance = 3
  
  # settings for new-gen strategy
  # Maximum number of shards that will be rebalanced in one rebalance round.
  # The lower of this and `relative-limit` will be used.
  absolute-limit = 100

  # Maximum number of shards that will be rebalanced in one rebalance round.
  # Fraction of total number of (known) shards.
  # The lower of this and `absolute-limit` will be used.
  relative-limit = 0.1
}
```

Reply:
I agree. The `strategy2` stuff is too ugly. I have reworked the deprecation story. Now the choice is made by `least-shard-allocation-strategy.rebalance-absolute-limit` > 0. Later we will change the default of this to enable the new algorithm by default. In release notes and reference docs we will recommend setting this property to use the new algorithm.